### DISCUSSION TITLE
Cuda out of memory

### DISCUSSION THREAD

**Discussion by skylning** [opengeos/segment-geospatial#155] [Q&A] [answered]
when i run the examples,i get a error :
torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 72.26 GiB (GPU 0; 23.99 GiB total capacity; 3.29 GiB already allocated; 18.02 GiB free; 3.35 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

and my code as follow:
image =test.tif
sam = SamGeo(
    model_type="vit_h",
    checkpoint='sam_vit_h_4b8939.pth',
    sam_kwargs=None,
)
sam.generate(image, output="masks.tif", foreground=True, unique=True)
sam.show_masks(cmap="binary_r")
sam.show_anns(axis="off", alpha=1, output="annotations.tif")
sam.tiff_to_vector("masks.tif", "masks.gpkg")

**Comment by giswqs** [comment, marked as answer]
What is the file size of your image? Try a small image before applying it on a large image.

**Comment by oscarbau** [comment]
> when i run the examples,i get a error : torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 72.26 GiB (GPU 0; 23.99 GiB total capacity; 3.29 GiB already allocated; 18.02 GiB free; 3.35 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation. See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF
> 
> and my code as follow: image =test.tif sam = SamGeo( model_type="vit_h", checkpoint='sam_vit_h_4b8939.pth', sam_kwargs=None, ) sam.generate(image, output="masks.tif", foreground=True, unique=True) sam.show_masks(cmap="binary_r") sam.show_anns(axis="off", alpha=1, output="annotations.tif") sam.tiff_to_vector("masks.tif", "masks.gpkg")

Did you find any solution apart from using a smaller image?
